### PR TITLE
#3091 Pss1a, Pss2a and other necessary models

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv
Comment (joyelfeghali) [inline nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv:1]: nrt fails for this test case. We have to modify it when we add st4b, there is no modification of PssIEEE2B

File: nrt/data/SMIB/Standard/TestCase3ST4B/reference/outputsTestCase3ST4B/curves/curves.csv
Comment (joyelfeghali) [inline nrt/data/SMIB/Standard/TestCase3ST4B/reference/outputsTestCase3ST4B/curves/curves.csv:1]: same for this nrt

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo [hunk 1]
```diff
@@ -16,41 +16,50 @@ model LeadMOrderLag "Lead-lag filter with M poles"
   extends Modelica.Blocks.Interfaces.SISO(y(start = Y0));
 
   parameter Real K = 1 "Gain";
-  parameter Integer M = 1 "Lag Order";
-  parameter Types.Time t1 "Lead time constant";
-  parameter Types.Time t2 "Lag time constant";
-  parameter Real Y0 = 0 "Initial or guess value of output" annotation(
-  Dialog(group="Initialization"));
+  parameter Integer M "Lag order";
+  parameter Integer MMax = 6 "Maximum lag order";
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo:20]: We need a fixed number of first order filters known before compilation, hence the parameter MMax.

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo [hunk 2]
```diff
@@ -16,41 +16,50 @@ model LeadMOrderLag "Lead-lag filter with M poles"
...
     Line(points = {{-120, 0}, {-62, 0}}, color = {0, 0, 127}));
+  connect(leadlag.y, firstOrderCascade[1].u);
 
-  if M > 1 then
-    connect(leadlag.y, firstOrderCascade[1].u);
-    connect(firstOrderCascade[M-1].y, y);
-  else
-    connect(leadlag.y, y);
-  end if;
+  for i in 1:MMax-2 loop
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo:41]: It is assumed that MMax is at least 3. I can write it somewhere.

File: nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv
Comment (ErwanGuichard49) [reply nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv:1]: Strangely, the nrt was successful on my computer.

File: nrt/data/SMIB/Standard/TestCase3ST4B/reference/outputsTestCase3ST4B/curves/curves.csv
Comment (ErwanGuichard49) [reply nrt/data/SMIB/Standard/TestCase3ST4B/reference/outputsTestCase3ST4B/curves/curves.csv:1]: Successful on my computer.

File: nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv
Comment (joyelfeghali) [reply nrt/data/SMIB/Standard/TestCase1ST4B/reference/outputsTestCase1ST4B/curves/curves.csv:1]: Why the LeadMOrderLag would give different values?

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/RampTrackingFilter.mo [hunk 3]
```diff
@@ -16,43 +16,52 @@ model RampTrackingFilter "Ramp tracking filter"
...
 
   parameter Real K = 1 "Gain";
-  parameter Integer M = 1 "M Order";
-  parameter Integer N = 1 "N Order";
-  parameter Types.Time t1 "Lead time constant";
-  parameter Types.Time t2 "Lag time constant";
-  parameter Real Y0 = 0 "Initial or guess value of output" annotation(
-  Dialog(group="Initialization"));
+  parameter Integer M "Lag order";
+  parameter Integer N "Order";
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/RampTrackingFilter.mo:20]: Indeed, the default value is overridden by the value set in the .par file. I have restored the default values for M and N.

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/LeadMOrderLag.mo:41]: Another possibility is to turn MMax and NMax into final parameters, as their value is fixed from the compilation and thus cannot be changed with the .par file. The only way to increase MMax and NMax is to change the value in the .mo file and to recompile.

File: dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsTGov1SexsPss2A.xml [hunk 4]
```diff
@@ -17,7 +17,7 @@
     <dyn:unitDynamicModel id="generator" name="Dynawo.Electrical.Machines.OmegaRef.GeneratorSynchronous" initName="Dynawo.Electrical.Machines.OmegaRef.GeneratorSynchronousExt4W_INIT"/>
     <dyn:unitDynamicModel id="voltageRegulator" name="Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.SEXS" initName="Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.SEXS_INIT"/>
     <dyn:unitDynamicModel id="governor" name="Dynawo.Electrical.Controls.Machines.Governors.Standard.Steam.TGOV1" initName="Dynawo.Electrical.Controls.Machines.Governors.Governor_INIT"/>
-    <dyn:unitDynamicModel id="pss" name="Dynawo.Electrical.Controls.Machines.PowerSystemStabilizers.Standard.PSS2A" initName="Dynawo.Electrical.Controls.Machines.PowerSystemStabilizers.BaseClasses_INIT.BasePss_INIT"/>
+    <dyn:unitDynamicModel id="pss" name="Dynawo.Electrical.Controls.Machines.PowerSystemStabilizers.Standard.PSS2A1" initName="Dynawo.Electrical.Controls.Machines.PowerSystemStabilizers.BaseClasses_INIT.BasePss_INIT"/>
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsTGov1SexsPss2A.xml:20]: Done

**Review by joyelfeghali — APPROVED** [review]

**Pull request merged**